### PR TITLE
Add upcoming rollups logic

### PR DESCRIPTION
**Pull request by antooni**
This PR implements logic to show/hide upcoming rollups.

The design is not yet fully ready and will be aligned in the next PR. 

This feature is hidden behind the feature flag on production

Resolves L2B-977

### CODE REVIEW

**Review by vitali-l2beat — APPROVED** [review]

File: packages/frontend/src/components/project/UpcomingBar.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    >
+      <span className="flex flex-row">
+        <span className="flex items-center">
+          <UpcomingIcon />
+        </span>
+        <span className="ml-2 text-base font-medium dark:text-white">
+          <span className="md:hidden">This is an upcoming project</span>
+          <span className="hidden md:block">
+            This is an upcoming project. Stay tuned!
+          </span>
```
Comment (michalsidzej) [inline packages/frontend/src/components/project/UpcomingBar.tsx:22]:
```suggestion
          This is an upcoming project.{" "}
          <span className="hidden md:block">
            Stay tuned!
          </span>
```
Nitpick and probably not necessary

File: packages/frontend/src/components/table/filters/checkboxes/IncludeUpcomingCheckbox.tsx [hunk 2]
```diff
@@ -11,7 +12,7 @@ export function IncludeUpcomingCheckbox({
 }: IncludeUpcomingCheckboxProps) {
   return (
     <Checkbox
-      className={className}
+      className={cx('bg-purple-100 text-white dark:bg-purple-100', className)}
```
Comment (michalsidzej) [inline packages/frontend/src/components/table/filters/checkboxes/IncludeUpcomingCheckbox.tsx:15]:
```suggestion
      className={cx('bg-purple-100 text-white', className)}
```

File: packages/frontend/src/pages/scaling-tvl/view/ScalingTvlView.tsx [hunk 3]
```diff
@@ -64,14 +65,16 @@ export function ScalingTvlView({
...
+        !project.isUpcoming && <NumberCell>{project.tvl}</NumberCell>,
     },
     {
       name: '7d Change',
       tooltip: 'Change in the total value locked as compared to a week ago.',
       alignRight: true,
       getValue: (project) =>
-        !project.isArchived && (
+        !project.isArchived &&
+        !project.isUpcoming && (
```
Comment (michalsidzej) [inline packages/frontend/src/pages/scaling-tvl/view/ScalingTvlView.tsx:77]: Maybe we create a variable `isActive`?

File: packages/frontend/src/scripts/configureFilters.ts [hunk 4]
```diff
@@ -17,8 +20,6 @@ export function configureFilters() {
   const archivedL2s = archived.filter((x) => L2s.includes(x))
   const activeL2s = L2s.filter((x) => !archivedL2s.includes(x))
 
-  console.log(activeL2s.length)
```
Comment (michalsidzej) [inline packages/frontend/src/scripts/configureFilters.ts:20]: Oh, hello

File: packages/frontend/src/scripts/configureFilters.ts [hunk 5]
```diff
@@ -49,4 +50,10 @@ export function configureFilters() {
       )
     })
   })
+
+  upcomingCheckbox?.addEventListener('change', () => {
+    upcoming.forEach((x) => {
+      x.classList.toggle('hidden')
```
Comment (michalsidzej) [inline packages/frontend/src/scripts/configureFilters.ts:56]:
```suggestion
      x.classList.toggle('hidden', !upcomingCheckbox.checked)
```

File: packages/frontend/src/utils/orderByTvl.ts [hunk 6]
```diff
@@ -1,20 +1,23 @@
...
     tvlApiResponse.projects[project.id.toString()]?.charts.hourly.data.at(
       -1,
     )?.[1] ?? 0
 
   const sortByTvl = (a: T, b: T) => getTvl(b) - getTvl(a)
 
-  return [...active].sort(sortByTvl).concat([...archived].sort(sortByTvl))
+  return [...active]
+    .sort(sortByTvl)
+    .concat(...upcoming)
```
Comment (michalsidzej) [inline packages/frontend/src/utils/orderByTvl.ts:21]: So we don't sort upcoming? Alphabetical order? Or random?

Comment (antooni) [reply packages/frontend/src/utils/orderByTvl.ts:21]: I am not sure just for now, lets leave for the next PR

**Pull request merged**